src/classes/BDI_DataImport_CTRL.cls

```diff
@@ -86,72 +84,23 @@ public with sharing class BDI_DataImport_CTRL {
         p.setRedirect(true);
         return p;
     }
-
-    private class BDICtrlException extends Exception {}
     
     public PageReference importData() {
-        try {
-        	// validate settings
-        	if (diSettings.Contact_Custom_Unique_ID__c != null) {
-                // make sure c1 and c2 have the appropriately named field
-        	    if (!UTIL_Describe.isValidField(UTIL_Namespace.StrTokenNSPrefix('DataImport__c'), 'Contact1_' + diSettings.Contact_Custom_Unique_ID__c) ||
-        	       !UTIL_Describe.isValidField(UTIL_Namespace.StrTokenNSPrefix('DataImport__c'), 'Contact2_' + diSettings.Contact_Custom_Unique_ID__c)) {
-                    throw(new BDICtrlException(string.format(Label.bdiContactCustomIdError, 
-                        new string[]{diSettings.Contact_Custom_Unique_ID__c, 
-                        	'Contact1_' + diSettings.Contact_Custom_Unique_ID__c,
-                        	'Contact2_' + diSettings.Contact_Custom_Unique_ID__c})));
-        	    }
-        	}
-        	
-            if (diSettings.Account_Custom_Unique_ID__c != null) {
-                // make sure a1 and a2 have the appropriately named field
-                if (!UTIL_Describe.isValidField(UTIL_Namespace.StrTokenNSPrefix('DataImport__c'), 'Account1_' + diSettings.Contact_Custom_Unique_ID__c) ||
-                   !UTIL_Describe.isValidField(UTIL_Namespace.StrTokenNSPrefix('DataImport__c'), 'Account2_' + diSettings.Contact_Custom_Unique_ID__c)) {
-                    throw(new BDICtrlException(string.format(Label.bdiAccountCustomIdError, 
-                        new string[]{diSettings.Account_Custom_Unique_ID__c, 
-                            'Account1_' + diSettings.Account_Custom_Unique_ID__c,
-                            'Account2_' + diSettings.Account_Custom_Unique_ID__c})));
-                }
-            }
-            
-            // save any settings changes.
-            if(!Test.isRunningTest()) {
-	            try {
-	                upsert diSettings;
-	            } catch (Exception ex) {
-	                // this is a complete hack to work around the bug
-	                // https://github.com/SalesforceFoundation/Cumulus/issues/1094
-	                // for reasons I cannot fathom, we get an error trying to upsert the
-	                // settings object even though the UTIL_CustomSettingsFacade has already
-	                // done an upsert on it.  This failure only occurs if there is no org defined
-	                // custom setting instance.  The workaround is to catch the upsert failing, and
-	                // just clear the Id and do an insert.
-	                diSettings.Id = null;
-	                insert diSettings;
-	            }
-            }
-            
+    	if(!Test.isRunningTest())
+    	   upsert diSettings;
```

Comment:
Does this address your concert regarding reading settings directly from the Facade in the DataImport_API class?